programs/zstdcli.c

```diff
@@ -796,6 +798,57 @@ int main(int argCount, const char* argv[])
                         continue;
                     }
 #endif
+
+                    if (longCommandWArg(&argument, "--filelist=")) {
+                        /* note : in theory, it's better to just store the arguments at this stage,
+                         *    and only start to load & interpret the file after command line is parsed.
+                         *    For a single file, it would be easy to just store its name here, and parse later.
+                         *    However, this implementation makes it possible to read multiple files.
+                         *    An equivalent will have to be able to store multiple file names.
+                         */
+                        FileNamesTable* extendedTable;
+                        FileNamesTable* curTable;
+                        FileNamesTable* concatenatedTables;
+
+                        if (!UTIL_fileExist(argument) || !UTIL_isRegularFile(argument)){
+                            DISPLAYLEVEL(1, "[ERROR] wrong fileName: %s\n", argument);
+                            CLEAN_RETURN(badusage(programName));
+                        }
+
+                        extendedTable = UTIL_createFileNamesTable_fromFileName(argument);
+                        if (!extendedTable) {
+                            CLEAN_RETURN(badusage(programName));
+                        }
+
+
+                        filenameTable[filenameIdx] = NULL;   /* marking end of table */
+                        filenameIdx += (unsigned) extendedTable->tableSize;
+
+                        curTable = UTIL_createFileNamesTable(filenameTable, filenameTableSize, tableBuf);
+
+                        if (!curTable) {
+                            UTIL_freeFileNamesTable(extendedTable);
+                            CLEAN_RETURN(badusage(programName));
+                        }
+
+                        concatenatedTables = UTIL_concatenateTwoTables(curTable, extendedTable);
+                        if (!concatenatedTables) {
+                            UTIL_freeFileNamesTable(curTable);
+                            UTIL_freeFileNamesTable(extendedTable);
+                            CLEAN_RETURN(badusage(programName));
+                        }
+
+                        /* transfer ownership */
+                        filenameTable = concatenatedTables->fileNames;
+                        filenameTableSize = concatenatedTables->tableSize;
```

Comment:
I think at this point we've lost the guarantee that we have enough space for all the filenames. So we could segfaults on line 1034 of the updated diff. If that is correct, can we add a test case that exposes this bug?

https://github.com/facebook/zstd/blob/6d5e0f1c9f57dd74edc5ccb4a96c0bbb5de96f7d/programs/zstdcli.c#L981

Reply:
Actually, it looks like we won't segfault, since we allocate a table of the correct size, `filenameTableSize` is wrong though.

Reply:
Well, I think it would segfault if we had 2 input files, since we use `filenameTableSize` to determine the table size.

Reply:
Yep, segfault generated.
This patch is not ready for merge.